src/helpers/chain.js

```diff
@@ -0,0 +1,379 @@
+
+      // If it is an offer for the round we are checking for
+      if (offerRound === roundId) {
+        // If there is no error
+        if (!currentOffer["status"].hasOwnProperty("error")) {
+          return true;
+        }
+      }
+
+      /**
+       * Else if offer round id is less than the round we want to check and its 
+       * satisfied
+       */
+      else if (
+        offerRound < roundId &&
+        !currentOffer["status"].hasOwnProperty("error")
+      ) {
+
+        /**
+         * Return false because there cannot be a submission for a newer round 
+         * before this offer
+         */
+        return false;
+      }
+    }
+  }
+  return false;
+};
+
+/**
+ * Function to query price from chain
+ * @param {string} feed feed name of the price to query in the form of ATOM-USD
+ * @returns {number} the latest price
+ */
+export const queryPrice = async (feed) => {
+  try {
+    // Read value from vstorage
+    const capDataStr = await readVStorage(feed, false);
+
+    //parse the value
+    let capData = JSON.parse(JSON.parse(capDataStr).value);
+    capData = JSON.parse(capData.values[0]);
+    // Replace any extra characters
+    capData = JSON.parse(capData.body.replaceAll("\\", ""));
+
+    // Get the latest price by dividing amountOut by amountIn
+    let latestPrice =
+      Number(capData.amountOut.value.digits) /
+      Number(capData.amountIn.value.digits);
+
+    console.log(feed + " Price Query: " + String(latestPrice));
+    return latestPrice;
+  } catch (err) {
+    console.log("ERROR querying price", err);
+    return -1;
+  }
+};
+
+/**
+ * Function to get oracles feed invitations
+ * @returns {Object} an object containing feed invitation IDs. Each field in *                   the object represents the feed name (Ex. ATOM-USD) and its *                   value is a number * where which is the invitation ID.
+ */
+export const getOraclesInvitations = async () => {
+  let { agoricNames, fromBoard, vstorage } = await makeRpcUtils({ fetch });
+
+  let feedBoards = agoricNames.reverse;
+
+  let feedInvs = {};
+
+  const current = await getCurrent(String(FROM), fromBoard, { vstorage });
+  const invitations = current.offerToUsedInvitation;
+
+  // For each invitation
+  for (let inv in invitations) {
+    let boardId = invitations[inv].value[0].instance.boardId;
+    let feed = feedBoards[boardId].split(" price feed")[0];
+
+    feedInvs[feed] = Number(inv);
+  }
+
+  return feedInvs;
+};
+
+/**
+ * Function to query round from chain
+ * @param {string} feed feed name of the price to query in the form of ATOM-USD
+ * @returns {Object} the latest round
+ * @returns {number} returns.round_id The round id
+ * @returns {number} returns.started_at The timestamp when the round *                   was started
+ * @returns {string} returns.started_by The address of who started the round
+ * @returns {boolean} returns.submission_made Whether a submission to this *                    round was made by the oracle
```

Comment:
```suggestion
 * @returns {boolean} returns.submission_made Whether a submission to this 
 *                    round was made by the oracle
```

Reply:
Fixed in https://github.com/jacquesvcritien/agoric-cl-middleware/commit/9f027c8cb34abf5790b78708b269ee97944c051e